modelbaker/iliwrapper/ili2dbconfig.py

```diff
@@ -111,31 +111,35 @@ def ilidata_directories(self):
 
 
 class Ili2DbCommandConfiguration:
-    def __init__(self):
-        self.base_configuration = BaseConfiguration()
-
-        self.dbport = ""
-        self.dbhost = ""
-        self.dbpwd = ""
-        self.dbusr = ""
-        self.dbauthid = ""
-        self.db_use_super_login = False
-        self.database = ""
-        self.dbschema = ""
-        self.dbfile = ""
-        self.dbservice = None
-        self.sslmode = None
-        self.tool = None
-        self.ilifile = ""
-        self.ilimodels = ""
-        self.tomlfile = ""
-        self.dbinstance = ""
-        self.db_odbc_driver = ""
-        self.disable_validation = False
-        self.metaconfig = None
-        self.metaconfig_id = None
-        self.metaconfig_params_only = False
-        self.db_ili_version = None
+    def __init__(self, other=None):
+        if not isinstance(other, Ili2DbCommandConfiguration):
```

Comment:
I understand the approach but why do we do that? Ain't it better to have all the "default" values initialized here?

Reply:
Yes, members are still being initialized to default values if `other` is not passed.

This is just to support creating e.g., a _DeleteConfiguration_ (or any subclass object) from an existing _Ili2DbCommandConfiguration_. That is, to get all values from the existing _Ili2DbCommandConfiguration_ to the new _DeleteConfiguration_ without having to copy them manually or going to other parts of the code to initialize members that correspond to _Ili2DbCommandConfiguration_, which could be a pain and in fact error prone.

We could've also done something like: First initialize all members to default values, and only then, see if we have an `other` object, and if so, get values from it, overriding the default ones. The current implementation avoids setting the same member twice, that's why I chose it.

Does it make sense?

Reply:
okay thanks for clarification